src/controllers/product.controller.js

```diff
@@ -1,27 +1,20 @@
 import { collectionServices, productsServices } from '../services';
+import { asyncWrapper } from '../helpers';
 
-const CreateCollection = async (req, res) => {
-  try {
-    const collectionName = req.body.name;
-    const userId = req.user.id;
-    const collectionObj = {
-      userId,
-      name: collectionName,
-    };
-    const collection = await collectionServices.createCollection(collectionObj);
-    return res.status(201).json({
-      code: 201,
-      message: 'Collection Created.',
-      collection,
-    });
-  } catch (error) {
-    return res.status(500).json({
-      code: 500,
-      message: 'Internal Error.',
-      error,
-    });
-  }
-};
+const CreateCollection = asyncWrapper(async (req, res) => {
+  const collectionName = req.body.name;
+  const userId = req.user.id;
+  const collectionObj = {
+    userId,
+    name: collectionName,
+  };
+  const collection = await collectionServices.createCollection(collectionObj);
+  return res.status(201).json({
+    code: 201,
+    message: 'Collection Created.',
+    collection,
+  });
+});
```

Comment:
Good for introducing the async wrapper, but we can make the code more readable by moving it into the routes instead of controller or during the exports